app/tasks/maintenance_tasks/active_record_task.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module MaintenanceTasks
+  class ActiveRecordTask < Task
```

Comment:
This implementation makes the tradeoff that consumers need to know about the various subclasses, but we can make that fairly seamless with good generators and error messages.

We might even allow consumers to add their own generators (e.g. in an app with many custom tasks using the same custom enumerator builder). 🤷‍♂️